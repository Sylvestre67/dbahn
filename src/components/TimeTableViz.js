import React from 'react';
import PropTypes from 'prop-types';
import * as d3 from 'd3';

import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
	element: { height: '100%' },
});

class TimeTableViz extends React.Component {
	constructor(props) {
		super(props);
		this.element = React.createRef();
	}

	componentDidMount() {
		this.drawTracks();
	}

	drawTracks() {
		const { timetable, tracks } = this.props;

		console.log(timetable, tracks);

		const node = this.element.current;
		const el = d3.select(this.element.current);

		const margin = { top: 5, left: 30, bottom: 10, right: 0 },
			width = node.clientWidth,
			height = node.clientHeight;

		const svg = el
			.append('svg')
			.attr('width', width)
			.attr('height', height)
			.attr('class', 'tracks');

		svg.append('circle')
			.attr('r', 5)
			.attr('fill', 'blue');
	}

	updateTracks() {}

	render() {
		const { classes } = this.props;

		return <div className={classes.element} ref={this.element} />;
	}
}

export default withStyles(styles)(TimeTableViz);
