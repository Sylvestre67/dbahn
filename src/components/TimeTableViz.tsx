import React from 'react';
import * as d3 from 'd3';

import { withStyles, createStyles, WithStyles } from '@material-ui/core/styles';
import { Timetable } from '../context/__generated__/Timetable';

const styles = () =>
	createStyles({
		element: { height: '100%' },
	});

interface ITimeTableVizProps extends Timetable {}
interface ITimeTableVizProps extends WithStyles<typeof styles> {}

class TimeTableViz extends React.PureComponent<ITimeTableVizProps, {}> {
	private element = React.createRef<HTMLDivElement>();

	componentDidMount() {
		this.drawTracks();
	}

	drawTracks() {
		const { timetable, tracks } = this.props;
		let width: number, height: number;

		const node = this.element.current;
		const el = d3.select(this.element.current);

		const margin = { top: 5, left: 30, bottom: 10, right: 0 };

		if (node) {
			(width = node.clientWidth), (height = node.clientHeight);

			const svg = el
				.append('svg')
				.attr('width', width)
				.attr('height', height)
				.attr('class', 'tracks');

			svg.append('circle')
				.attr('r', 5)
				.attr('fill', 'blue');
		}
	}

	updateTracks() {}

	render() {
		const { classes } = this.props;
		return <div className={classes.element} ref={this.element} />;
	}
}

export default withStyles(styles)(TimeTableViz);
