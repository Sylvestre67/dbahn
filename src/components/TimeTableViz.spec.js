import React from 'react';
import { shallow } from 'enzyme';

import TimeTableViz from './TimeTableViz';
let component;

const props = {
	classes: { root: 'testing' },
};

describe('TimeTableViz', () => {
	beforeAll(() => {
		component = shallow(<TimeTableViz {...props} />);
	});

	it('should render correctly', () => {
		expect(component).toMatchSnapshot();
	});
});
