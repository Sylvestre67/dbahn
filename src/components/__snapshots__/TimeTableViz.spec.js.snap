// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeTableViz should render correctly 1`] = `
<TimeTableViz
  classes={
    Object {
      "element": "TimeTableViz-element-1",
      "root": "undefined testing",
    }
  }
/>
`;
